### PR TITLE
Deduplicate `nix` dependency versions

### PR DESCRIPTION
**Pull request by sholderbach**
# Description
Now we only use `nix 0.28.0`

Achieved by
- updating `ctrlc` to `3.4.4`
- updating `wl-clipboard-rs` to `0.8.1`
- update our own dependency on `nix` from `0.27` to `0.28`
  - required fixing uses of `nix::unistd::{tcgetpgrp,tcsetpgrp}`
  - now requires an I/O safe file descriptor
  - fake one pointing to `libc::STDIN_FILENO` (we were only accessing
`0` previously, dito for fish)

# User-Facing Changes
Better compile times and less to download as source dependencies

### CODE REVIEW

File: crates/nu-system/src/foreground.rs [hunk 1]
```diff
@@ -98,15 +101,28 @@ impl Drop for ForegroundChild {
...
 
+    /// Alternative to having to call `std::io::stdin()` just to get the file descriptor of stdin
+    ///
+    /// # Safety
+    /// I/O safety of reading from `STDIN_FILENO` unclear.
+    ///
+    /// Currently only intended to access `tcsetpgrp` and `tcgetpgrp` with the I/O safe `nix`
+    /// interface.
+    pub unsafe fn stdin_fd() -> impl AsFd {
+        unsafe { BorrowedFd::borrow_raw(libc::STDIN_FILENO) }
```
Comment (IanManske) [inline crates/nu-system/src/foreground.rs:123]: Oh cool, `tcgetpgrp` takes `impl AsFd` now. In that case, we can just use the standard lib `Stdin` type.

```suggestion
        std::io::stdin()
```

Comment (sholderbach) [reply crates/nu-system/src/foreground.rs:123]: The underlying [`impl AsFd for Stdin`](https://doc.rust-lang.org/src/std/os/fd/owned.rs.html#423-428) does the same, but unsure if having to create [`Stdin` via `stdin`](https://doc.rust-lang.org/nightly/src/std/io/stdio.rs.html#332-339) can be optimized out?

Comment (IanManske) [reply crates/nu-system/src/foreground.rs:123]: Oh ok, yeah it looks like it can't be [optimized out](https://godbolt.org/z/oxrrEo9hq).

Comment (sholderbach) [reply crates/nu-system/src/foreground.rs:123]: Thanks for godbolting! Funny how it completely extracts the constant into its `xor eax, eax` but is forced to keep the `call`

**Pull request merged**